src/components/FullDetailView/FullDetailPage.js

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+import React, { Component } from 'react'
+import { bindActionCreators } from 'redux'
+import { connect } from 'react-redux'
+import { Grid, Button } from 'react-bootstrap'
+import PropTypes from 'prop-types'
+import Modal from 'antd/lib/modal'
+import 'antd/dist/antd.css'
+
+import { uiInspectGetDefinition, uiInspectGetCuration, uiInspectGetHarvested, uiNavigation } from '../../actions/ui'
+import { curateAction } from '../../actions/curationActions'
+import { ROUTE_DEFINITIONS } from '../../utils/routingConstants'
+import EntitySpec from '../../utils/entitySpec'
+import Contribution from '../../utils/contribution'
+import ContributePrompt from '../ContributePrompt'
+import FullDetailComponent from './FullDetailComponent'
+
+/**
+ * Component that renders the Full Detail View as a Page or as a Modal
+ * based on modalView property
+ *
+ */
+export class FullDetailPage extends Component {
+  componentDidMount() {
+    const { path, uiNavigation, component } = this.props
+
+    if (path && component) {
+      this.handleNewSpec(component)
+    }
+    uiNavigation({ to: ROUTE_DEFINITIONS })
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const { path, component } = nextProps
+
+    if (path && path !== this.props.path) {
+      this.handleNewSpec(component)
+    }
+  }
+
+  /**
+   * Get the data for the current definition
+   *
+   */
+  handleNewSpec = component => {
+    const { token, uiInspectGetDefinition, uiInspectGetCuration, uiInspectGetHarvested } = this.props
+    if (component) {
+      uiInspectGetDefinition(token, component)
+      uiInspectGetCuration(token, component)
+      uiInspectGetHarvested(token, component)
+    }
+  }
+
+  /**
+   * Dispatch the action to save a contribution
+   *
+   */
+  doContribute = description => {
+    const { token, component } = this.props
+    const patches = Contribution.buildContributeSpec({}, component)
+    const spec = { description: description, patches }
+    curateAction(token, spec)
+  }
+
+  /**
+   * Shows the Modal to save a Contribution
+   *
+   */
+  doPromptContribute = () => {
+    const { component } = this.props
+    if (!Contribution.hasChange(component)) return
+    this.refs.contributeModal.open()
+  }
+
+  handleSave = changes => {
+    const { onSave, component } = this.props
+    const newComponent = { ...component, changes }
+    onSave && onSave(component, newComponent)
+  }
+
+  handleClose = () => {
+    const { onClose } = this.props
+    onClose()
+  }
+
+  render() {
+    const { path, component, definition, curation, harvest, modalView, visible } = this.props
+
+    return modalView ? (
+      <Modal
```

Comment:
consider breaking this out into the following just to make the logic clearer

```javascript
return modalView ? this.renderModal() : this renderStandard()
```

or some such.

Reply:
Actually my comment here would be addressed if this were actually two different components.